server/src/domain/library/library.service.ts

```diff
@@ -66,11 +70,61 @@ export class LibraryService {
       config.library.scan.enabled,
     );
 
+    this.watchEnabled = config.library.watch.enabled;
+
+    for (const library of await this.repository.getAll(true, LibraryType.EXTERNAL)) {
+      if (library.watched) {
+        await this.watch(library);
+      }
+    }
+
     this.configCore.config$.subscribe((config) => {
       this.jobRepository.updateCronJob('libraryScan', config.library.scan.cronExpression, config.library.scan.enabled);
     });
   }
 
+  async teardown() {
+    this.watcher?.close();
+  }
+
+  async watch(library: LibraryEntity) {
+    if (!this.watchEnabled) {
+      return;
+    }
+
+    const extensions = mimeTypes.getSupportedFileExtensions();
+
+    this.logger.log(`Watching ${library.importPaths} for ${extensions} files`);
```

Comment:
This will be a pretty big log message, right? Do we need to print out all the extensions?